lib/engines/content_object_store/app/controllers/content_object_store/content_block_editions_controller.rb

```diff
@@ -12,7 +12,18 @@ def new
       @schemas = ContentObjectStore::ContentBlockSchema.all
     else
       @schema = ContentObjectStore::ContentBlockSchema.find_by_block_type(params[:block_type].underscore)
-      @content_block_edition = ContentObjectStore::ContentBlockEdition.new
+
+      form_attributes = @schema.fields.each_with_object({}) do |field, hash|
+        hash[field] = nil
+        hash
+      end
```

Comment:
I think I'd feel happier with the logic being inside the Form object. That way we keep from overloading the controller with too much logic and can unit test a bit more reliably. We could potentially have two form objects that inherit from `ContentObjectStore::ContentBlockEditionForm` with the correct values for each, so `ContentObjectStore::ContentBlockEditionForm` contains the base logic:

```ruby
class ContentObjectStore::ContentBlockEditionForm
  include ContentObjectStore::Engine.routes.url_helpers
  
  attr_reader :content_block_edition, :schema

  def initialize(content_block_edition, schema)
    # blah
  end
end
```

And we have two subclasses that contain the rules for edit/update:

```ruby
class ContentObjectStore::ContentBlockEditionForm::Update < ContentObjectStore::ContentBlockEditionForm
   def attributes
     @schema.fields.each_with_object({}) do |field, hash|
         hash[field] = nil
         hash
       end
     end

    def back_path
       content_object_store.content_object_store_content_block_editions_path
    end
end 
```

```ruby
class ContentObjectStore::ContentBlockEditionForm::Edit < ContentObjectStore::ContentBlockEditionForm
   def attributes
     @schema.fields
   end

    def back_path
       content_object_store.content_object_store_content_block_editions_path
    end
end 
```

Happy to chat this through if you want 👍

Reply:
Yeah that makes sense 👍

Reply:
Yeah, that's fair, I was trying to think of a more elegant way to specify create/update without a (somewhat ugly) `form_type` parameter

Reply:
I've been using Thoughtbot's Ruby Science book a lot recently to get me into the latest and greatest Ruby practice, so was inspired in part by this https://thoughtbot.com/ruby-science/replace-conditional-with-polymorphism.html